File: go/longestStreak/solution_test.go
package solution

import "testing"

func TestCaseOne(t *testing.T) {
	a := NewNode(5)
	b := NewNode(5)
	c := NewNode(7)
	d := NewNode(7)
	e := NewNode(7)
	f := NewNode(6)

	a.next = b
	b.next = c
	c.next = d
	d.next = e
	e.next = f

	want := 3
	got := longestStreak(a)

	if got != want {
		t.Errorf("got %d, want %d", got, want)
	}
}

func TestCaseTwo(t *testing.T) {
	a := NewNode(3)
	b := NewNode(3)
	c := NewNode(3)
	d := NewNode(3)
	e := NewNode(9)
	f := NewNode(9)

	a.next = b
	b.next = c
	c.next = d
	d.next = e
	e.next = f

	want := 4
	got := longestStreak(a)

	if got != want {
		t.Errorf("got %d, want %d", got, want)
	}
}

func TestCaseThree(t *testing.T) {
	a := NewNode(9)
	b := NewNode(9)
	c := NewNode(1)
	d := NewNode(9)
	e := NewNode(9)
	f := NewNode(9)

	a.next = b
	b.next = c
	c.next = d
	d.next = e
	e.next = f

	want := 3
	got := longestStreak(a)

	if got != want {
		t.Errorf("got %d, want %d", got, want)
	}
}

func TestCaseFour(t *testing.T) {
	a := NewNode(5)
	b := NewNode(5)

	a.next = b

	want := 2
	got := longestStreak(a)

	if got != want {
		t.Errorf("got %d, want %d", got, want)
	}
}

func TestCaseFive(t *testing.T) {
	want := 0
	got := longestStreak(nil)

	if got != want {
		t.Errorf("got %d, want %d", got, want)
	}
}

File: py/is_prime/solution_test.py
import unittest
from solution import Solution

class TestIsPrime(unittest.TestCase):
    def setUp(self):
        self.solution = Solution()

    def test_case_one(self):
        result = self.solution.is_prime(7)
        self.assertTrue(result)

    def test_case_two(self):
        result = self.solution.is_prime(3)
        self.assertTrue(result)

    def test_case_three(self):
        result = self.solution.is_prime(4)
        self.assertFalse(result)

    def test_case_four(self):
        result = self.solution.is_prime(5)
        self.assertTrue(result)

    def test_case_five(self):
        result = self.solution.is_prime(6)
        self.assertFalse(result)

    def test_case_six(self):
        result = self.solution.is_prime(7)
        self.assertTrue(result)

    def test_case_seven(self):
        result = self.solution.is_prime(8)
        self.assertFalse(result)

    def test_case_eight(self):
        result = self.solution.is_prime(25)
        self.assertFalse(result)

    def test_case_nine(self):
        result = self.solution.is_prime(31)
        self.assertTrue(result)

    def test_case_ten(self):
        result = self.solution.is_prime(2017)
        self.assertTrue(result)

    def test_case_eleven(self):
        result = self.solution.is_prime(2048)
        self.assertFalse(result)

    def test_case_twelve(self):
        result = self.solution.is_prime(1)
        self.assertFalse(result)

    def test_case_thirteen(self):
        result = self.solution.is_prime(713)
        self.assertFalse(result)
